File: legacy-rust/src/aquery/query_result.rs
use serde::Deserialize;

/// Proto Scheme based on the following
/// https://github.com/bazelbuild/bazel/blob/master/src/main/protobuf/analysis_v2.proto
#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct QueryResult {
    pub artifacts: Vec<Artifact>,
    pub actions: Vec<Action>,
    pub targets: Vec<Target>,
    pub rule_classes: Vec<RuleClass>,
    pub dep_set_of_files: Vec<DepSetOfFiles>,
    pub path_fragments: Vec<PathFragment>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Artifact {
    pub id: u32,
    pub path_fragment_id: u32,
    pub is_tree_artifact: Option<bool>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Action {
    pub target_id: u32,
    pub action_key: String,
    pub mnemonic: String,
    pub configuration_id: u32,
    pub arguments: Vec<String>,
    pub environment_variables: Vec<EnvironmentVariable>,
    pub input_dep_set_ids: Vec<u32>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Target {
    pub id: u32,
    pub label: String,
    pub rule_class_id: u32,
}

#[derive(Debug, Deserialize)]
pub struct RuleClass {
    pub id: u32,
    pub name: String,
}

#[derive(Debug, Deserialize)]
pub struct EnvironmentVariable {
    pub key: String,
    pub value: String,
}

#[derive(Debug, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PathFragment {
    pub id: u32,
    pub label: String,
    pub parent_id: Option<u32>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DepSetOfFiles {
    pub id: u32,
    pub direct_artifact_ids: Option<Vec<u32>>,
    pub transitive_dep_set_ids: Option<Vec<u32>>,
}

File: Sources/ActionQuery/QueryResult.swift
import Foundation

// MARK: - Bazel Query Result Types

/// Proto Scheme based on the following
/// https://github.com/bazelbuild/bazel/blob/master/src/main/protobuf/analysis_v2.proto
package struct QueryResult: Codable {
    package let artifacts: [Artifact]
    package let actions: [Action]
    package let targets: [Target]
    package let ruleClasses: [RuleClass]
    package let depSetOfFiles: [DepSetOfFiles]
    package let pathFragments: [PathFragment]
}

package struct Artifact: Codable {
    package let id: UInt32
    package let pathFragmentId: UInt32
    package let isTreeArtifact: Bool?
}

package struct Action: Codable {
    package let targetId: UInt32
    package let actionKey: String
    package let mnemonic: String
    package let configurationId: UInt32
    package let arguments: [String]
    package let environmentVariables: [EnvironmentVariable]
    package let inputDepSetIds: [UInt32]
}

package struct Target: Codable {
    package let id: UInt32
    package let label: String
    package let ruleClassId: UInt32
}

package struct RuleClass: Codable {
    package let id: UInt32
    package let name: String
}

package struct EnvironmentVariable: Codable {
    package let key: String
    package let value: String
}

package struct PathFragment: Codable {
    package let id: UInt32
    package let label: String
    package let parentId: UInt32?
}

package struct DepSetOfFiles: Codable {
    package let id: UInt32
    package let directArtifactIds: [UInt32]?
    package let transitiveDepSetIds: [UInt32]?
}
